### ISSUE TITLE
Command that fail if no output is connected can fail on startup due to modeset delay

### ISSUE THREAD

**Issue by kennylevinsen** [swaywm/sway#8433]
A handful of commands have the following construct:

```c
		if (config->reading || !config->active) {
			return cmd_results_new(CMD_DEFER, NULL);
		} else if (!root->outputs->length) {
			return cmd_results_new(CMD_INVALID,
					"Can't run this command while there's no outputs connected.");
		}
```

These commands expect that once the deferred configuration is run, any outputs that we were expecting would have appeared. This is no longer true due to the modeset delay timer.

We can fix this three ways:

1. Remove the output requirement
2. Delay deferred commands further to happen after modeset
3. Force-trigger an immediate modeset once the configuration is done.

**Comment by emersion** [comment]
> Remove the output requirement

Wouldn't the command then attempt to grab a `sway_output` or workspace, and then this fails because it's not been created yet?

**Comment by emersion** [comment]
Would make sense to me to trigger an immediate modeset right after startup or reload.

**Comment by Nefsen402** [comment]
I don't think special casing immediate modesets would fix this problem in the general case. It's possible I have a script that enables an output and then immediately changes a bunch of properties about it. It's not necessarily true that the only place we run into this is during configuration load/reload.

**Comment by emersion** [comment]
Hm right. What about delaying specifically only modesets triggered by `new_output` events?

**Comment by kennylevinsen** [comment]
> Wouldn't the command then attempt to grab a sway_output or workspace, and then this fails because it's not been created yet?

We always have a fallback output, so even though the focused output is very much not the one the user wants, all operations should be possible, no?

The only caveat is that focusing a particular output can't work before it's created. We might also want to have some special casing so that the fallback output doesn't end up being considered a preferred output to restore workspaces to and such.

But, fixing all that would be more work than just restoring the old behavior, even if it was always fragile, by doing a one-off immediate modeset. It does increase the likelihood of needing two modesets at startup though.

> It's possible I have a script that enables an output and then immediately changes a bunch of properties about it. 

Output configuration would just be stored and used when applicable, but trying to focus the output or move content to it immediately wouldn't work.

**Comment by philiprhoades** [comment]
I have this problem too - upgrading to 1.10.1 did not fix it so I am currently starting my 10 workspaces from a shell script . .

**Comment by emersion** [comment]
The fix is in master but not yet released.

**Comment by philiprhoades** [comment]
> The fix is in master but not yet released.

Excellent! - at least I learnt a little more as a result of the problem . . 

Thanks!
